[DOCS-10453] GitLab source code integration closed preview (#28378)
DataDog/documentation: content/en/integrations/guide/source-code-integration.md

| GitHub SaaS (github.com) | Yes | Yes |
| GitHub Enterprise Server | Yes | Yes |
| GitLab SaaS (gitlab.com) | Yes | Yes |

> | GitLab self-managed | Yes | Yes (Closed preview) |

Reviewer: ```suggestion
| GitLab self-managed | Yes | Yes (closed preview) |
```

{{% tab "GitLab" %}}

<div class="alert alert-warning">

> Repositories from GitLab instances are supported in closed preview. <a href="https://www.datadoghq.com/product-preview/gitlab-source-code-integration/">Join the preview</a>.

Reviewer: ```suggestion
Repositories from GitLab instances are supported in closed Preview. <a href="https://www.datadoghq.com/product-preview/gitlab-source-code-integration/">Join the Preview</a>.
```

Reviewer: Sorry - sent along my review and then remembered that we usually capitalize `Preview` 🤦🏻‍♀️

</div>

> To link telemetry with your source code, either install the GitLab integration (join the closed preview above) or upload your repository metadata with the [`datadog-ci git-metadata upload`][2] command. `datadog-ci v2.10.0` or later is required.

Reviewer: ```suggestion
To link telemetry with your source code, either install the GitLab integration (which requires joining the closed preview above) or upload your repository metadata with the [`datadog-ci git-metadata upload`][2] command. `datadog-ci v2.10.0` or later is required.
```
